src/net.ts

```diff
@@ -40,14 +50,8 @@ export default class Network {
 
   async checkProtocolVersion(): Promise<any> {
     const response = await this.provider.send('ain_checkProtocolVersion');
-    if (response.code === 1) {
-      const nodeProtoVer = response.protoVer;
-      if (semver.lte(VERSION_LIST[SDK_VERSION].min, nodeProtoVer) &&
-            (!VERSION_LIST[SDK_VERSION].max ||
-                semver.gte(VERSION_LIST[SDK_VERSION].max, nodeProtoVer))) {
-        console.log("Trying to adjust our protoVer to the node's..");
-        // Update protoVer if we can
-        this.protoVer = nodeProtoVer;
+    if (response.code === 1 && response.message !== PROTO_VER_INCOMPAT_ERROR) {
+      if (this.setProtoVer(response.protoVer)) {
         return this.provider.send('ain_checkProtocolVersion');
```

Comment:
As discussed offline, if it's not a clear case to retry with a new version, maybe this can be skipped this time.

Reply:
Removed!